src/crepe/system/InputSystem.h

```diff
@@ -13,6 +15,37 @@ namespace crepe {
 class Camera;
 class Button;
 class Transform;
+//! Event triggered when a button is clicked
+class ButtonPressEvent : public Event {
+public:
+	//! Metadata of the button.
+	const Metadata & meta_data;
```

Comment:
Metadata is één woord dus hier hoeft geen underscore tussen. Verander dit ook even in de constructor argumentnamen en de andere klassen.